src/document-submit.js

```diff
@@ -25,14 +26,18 @@ const PROP = '__AMP_SUBMIT';
 
 
 /**
- * @param {!Window} window
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @return {!Object}
  */
-export function installGlobalSubmitListener(window) {
-  if (!window[PROP]) {
-    window[PROP] = true;
-    window.document.documentElement.addEventListener(
-        'submit', onDocumentFormSubmit_, true);
-  }
+export function installGlobalSubmitListenerForDoc(ampdoc) {
+  return /** @type {!Object} */ (getServiceForDoc(ampdoc, 'submit', ampdoc => {
+    if (!ampdoc.win[PROP]) {
```

Comment:
Well, the code here didn't really change :)

What we need is:
1. Track whether the service is installed on ampdoc level, not window. I.e. there will be as many listeners as there ampdocs.
2. The listener should be installed on ampdoc's root node.
3. Why do we need `win[PROP]` here? `getServiceForDoc` already tests the existence.